src/csharp/Grpc.Core/ServerServiceDefinition.cs

```diff
@@ -45,6 +48,28 @@ private ServerServiceDefinition(Dictionary<string, IServerCallHandler> callHandl
             }
         }
 
+        /// <summary>
+        /// Returns a <see cref="Grpc.Core.ServerServiceDefinition" /> instance that
+        /// intercepts calls to the underlying service handler via the given interceptor.
+        /// This is an EXPERIMENTAL API.
+        /// </summary>
+        /// <param name="interceptor">The interceptor to register on service.</param>
+        public ServerServiceDefinition Intercept(Interceptor interceptor)
```

Comment:
a general thought on the Intercept(interceptor) methods:
there's a distinction between  the act of registering an interceptor    and   actually intercepting the call. In this PR, we are using  "Intercept" in the sense of "register interceptor", but the name "intercept" might have the feel of actually intercepting something.  Would renaming "InterceptWith(interceptor)" be clearer for the users? What does the Java API do here?
I dont' feel strongly about this, just wanted discuss the idea a bit.

Reply:
Java has `withInterceptor`, but I think it is pretty clear and more concise and C#y as is. The type of the receiver class should make it clear to the user what the operation being done is (i.e. Channel).